src/payment/cart/order-details/data/selectors.js

```diff
@@ -22,7 +22,7 @@ const paymentSelector = (state, props) =>
 
 const lastProductSelector = createSelector(
   paymentSelector,
-  ({ products }) => (products ? products[products.length - 1] : undefined),
+  ({ products }) => products[products.length - 1],
```

Comment:
The ternary operator here was handling a situation that never exists.  Removed it so I wouldn't have to write a test for the undefined clause.